fem/bilininteg.hpp

```diff
@@ -32,7 +32,10 @@ class BilinearFormIntegrator : public NonlinearFormIntegrator
 {
 protected:
    BilinearFormIntegrator(const IntegrationRule *ir = NULL)
-      : NonlinearFormIntegrator(ir) { }
+      : NonlinearFormIntegrator(ir), ceedDataPtr(NULL) { }
+
+   // CEED extension
+   CeedData* ceedDataPtr;
```

Comment:
Previously this was in all the individual integrators that supported Ceed (ie, `MassIntegrator`, `DiffusionIntegrator`, etc.), not in the extensions. In particular this is really an Integrator, and the extensions are for the `BilinearForm` itself. I am certainly open to a different design but it's not clear to me how to do it.

Reply:
The reason for this is that, in the current implementation, a `CeedOperator` is really an `Integrator` not a `BilinearForm`, since the merge of #1766 this data is actually in `NonlinearFormIntegrator`. I think it would be challenging to have only one `CeedOperator` for the whole `BilinearFormExtension`, but it might be possible using a `CompositeOperator`... I don't think this should be done in this PR though...

Reply:
There's already this (recently added with #1766): https://github.com/mfem/mfem/blob/b0770915511fc63fda3120b13c88db48946e4302/fem/nonlininteg.hpp#L32-L33
that is inherited by the `BilinearFormIntegrator` from the `NonlinearFormIntegrator`, so this is a second libCEED-related data member. What type of data is `ceedDataPtr` supposed to hold?

Reply:
`ceedDataPtr` is not added in this PR, it was there before. What you're seeing in the diff is this pointer being moved, but it's probably irrelevant because Yohann's #1766 restructured everything, changed this name, made it an object, and moved it again, so when master is merged in here everything will make sense.

Reply:
Okay, I thought `master` was already merged here (I didn't check).

It is probably best to merge it before further reviews because things will need to change anyway.